query-engine/query-engine-node-api/src/node_drivers/engine.rs

```diff
@@ -0,0 +1,140 @@
+use crate::engine::QueryEngine;
+
+#[allow(unused_imports)]
+use napi::bindgen_prelude::Promise as JsPromise;
+
+use napi::{Env, JsFunction, JsObject, JsString, JsUnknown};
+use napi_derive::napi;
+
+pub struct NodejsFunctionContext {
+    env: Env,
```

Comment:
Initially I wanted a `&'a Env`, but as `NodejsFunctionContext` is used in a struct with the `#[napi]` macro applied (namely, `QueryEngineNodeDrivers`), explicit lifetimes can't be used.